src/test/java/org/shikshalokam/backend/TestCRUDUserPermissions.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    @Test(description = "Validates the negative use case for CREATE PERMISSION ")
+    public void testCreatePermission_MissingRequiredFields() {
+        logger.info("Started calling --------- CreatePermission - Negative Test Missing Required Fields");
+
+        HashMap<String, Object> permissionMap = new HashMap<>();
+        permissionMap.put("code", "");
+        permissionMap.put("module", "");
+
+        // Example request type as a List (JSONObject handles this automatically)
+        List<String> requestTypeList = new ArrayList<>();
+        requestTypeList.add("POST");
+        permissionMap.put("request_type", requestTypeList);
+
+        permissionMap.put("api_path", "/mentoring/v1/entity/create");
+        permissionMap.put("status", "ACTIVE");
+
+        // Call the createPermission method
+        Response response = createPermission(permissionMap);
+        logger.info("Response Code: {}, \n Response Body: {}", response.getStatusCode(), response.getBody().asString());
+        assertEquals(response.getStatusCode(), 422, "Expected Bad Request for missing required fields - 422, but got " + response.getStatusCode());
+
+        String responseBody = response.getBody().asString();
+        assertTrue(responseBody.contains("Code field is empty") || responseBody.contains("Code is invalid, must not contain spaces")
+                || responseBody.contains("Module field is empty") || responseBody.contains("Module is invalid, must not contain spaces")
+                || responseBody.contains("request_type field is empty") || responseBody.contains("request_type is invalid, must be one of: GET,POST,PATCH,PUT,DELETE")
+                || responseBody.contains("API Path is invalid") || responseBody.contains("Status is invalid, must not contain spaces"),
+                "Expected error messages for missing title or invalid title not found in response: " + responseBody);
+
+        logger.info("Ended calling --------------- CreateUserRoles: Negative test completed with assertions on missing fields.");
+    }
+    @Test(description = "Fetches the list of available permissions already created")
+    public void getUserPermissions() {
+        logger.info("Started calling ---------- GET Permissions -------------");
+
+        Response response = given()
+                .header("X-auth-token", "bearer " + X_AUTH_TOKEN)
+                .when().get(getUserPermissionsEndPoint);
+
+        logger.info(response.getBody().asPrettyString());
+        assertEquals(response.getStatusCode(), 200, "Status code failed for getting permissions");
+
+        int permissionCount = response.jsonPath().getInt("result.results.count");
+        assertTrue(permissionCount > 0, "Expected permission count to be greater than zero");
+        logger.info("Total permissions fetched: " + permissionCount);
+
+        logger.info("Ended calling ---- GET Permissions: with all the assertions");
+    }
+
+    private URI createURI(String endpoint) {
```

Comment:
this move to the MentorBase class so that all projects can use the same .
public static URI createURI(String endpoint)

all the Test cases  can use like this  :
MentorBase.createURI("endpoint URL");